File: Viex.MyExpenses.Web/src/models/TransactionEntry.ts
import { CategoryDescriptor } from "./CategoryDescriptor"
import { TransactionType } from "./TransactionType"

export class TransactionEntry {
    
    transactionEntryId: number
    amount: number
    categoryId: number
    category: CategoryDescriptor
    date: Date
    description: string
    type: TransactionType

    constructor() {
        this.amount = 0
        this.date = new Date()
        this.type = TransactionType.expense
    }
}

File: Viex.MyExpenses.Persistence/Entities/TransactionEntry.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Viex.MyExpenses.Persistence.Entities
{
    public class TransactionEntry : BaseEntity
    {
        public long TransactionEntryId { get; set; }
        public decimal Amount { get; set; }
        public string Description { get; set; }

        public long CategoryId { get; set; }
        public CategoryDescriptor Category { get; set; }

        public long TypeId { get; set; }
        public TransactionTypeDescriptor Type { get; set; }
    }
}
